gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```diff
@@ -58,12 +67,32 @@ public void apply(Project rootProject) {
         JdksExtension jdksExtension = JdksPlugin.extension(rootProject, jdkDistributions);
 
         rootProject.getPluginManager().withPlugin("com.palantir.baseline-java-versions", unused -> {
-            rootProject
-                    .getExtensions()
-                    .getByType(BaselineJavaVersionsExtension.class)
-                    .getSetupJdkToolchains()
-                    .set(false);
+            BaselineJavaVersionsExtension baselineJavaVersionsExtension =
+                    rootProject.getExtensions().getByType(BaselineJavaVersionsExtension.class);
+            baselineJavaVersionsExtension.getSetupJdkToolchains().set(false);
+
+            jdksExtension.jdkMajorVersionsToUse().set(rootProject.provider(() -> Stream.of(
+                            jdksExtension.getDaemonTarget(),
+                            baselineJavaVersionsExtension.libraryTarget(),
+                            baselineJavaVersionsExtension.runtime().map(ChosenJavaVersion::javaLanguageVersion),
+                            baselineJavaVersionsExtension
+                                    .distributionTarget()
```

Comment:
These are all properties, so we really should get into the habit of `.zip`ing properties together just in case there's a task that is used to calculate the value of the property (and by doing `rootProject.provider` we lose that information).

Unfortunately this is really annoying to do as Gradle does not supply any sort of `Providers.zipWith(lots, of, different, values)` so best to hold off for now. I might make some util classes for this in gradle-utils.

```java
jdksExtension.getDaemonTarget()
```

Reply:
Actually, why not just have lots of `jdksExtension.jdkMajorVersionsToUse().add/addAll(` calls here for each one, rather than trying to combine them together? This should do the trick just fine.